thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/api/detection/AnomalyDetectionResource.java

```diff
@@ -464,6 +420,38 @@ TaskDTO generateTaskConfig(DetectionConfigDTO detectionConfigDTO,
     return taskDTO;
   }
 
+  long saveOnlineDetectionData(JsonNode payloadNode,
+      DatasetConfigDTO datasetConfigDTO, MetricConfigDTO metricConfigDTO)
+        throws JsonProcessingException {
+    JsonNode dataNode = payloadNode.get(DATA_FIELD);
+    String timeColumnName = datasetConfigDTO.getTimeColumn();
+    String datasetName = datasetConfigDTO.getDataset();
+    String metricName = metricConfigDTO.getName();
+
```

Comment:
@jasonyanwenl This code looks good.  Are there docs that tell users the formatting for adhoc data, or is that future work for Thirdeye?

Reply:
Thanks Harley! Yeah, the formatting for the ad-hoc data will be explained in the API doc. I am currently writing that. I will share it when I finished it.